### PR TITLE
✨ Initial implementation of clamp text

### PR DESCRIPTION
**Pull request by sparhami**
- Implement a single version of clamp, where the optional overflow
element is the last child of the component.
- Use overflow for clamping. Doing various optimizations (like using
Range) requires more work and has a lot of cases to consider. The
potential benefit is in the range of 5x-10x performance.
- Implement both a Shadow DOM and non-Shadow DOM version of the
component.
  * The tests need to be changed to have all of them run against both.

### CODE REVIEW

**Comment by honeybadgerdontcare** [comment]
When it's ready for review @alin04 for validator review

File: extensions/amp-clamp-text/validator-amp-clamp-text.protoascii [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  html_format: AMP
+  tag_name: "AMP-CLAMP-TEXT"
+  requires_extension: "amp-clamp-text"
+  attr_lists: "extended-amp-global"
+  attrs: {
+    name: "overflow-style"
+    value: "right"
+    value: "default"
+  }
+  spec_url: "https://www.ampproject.org/docs/reference/components/amp-clamp-text"
```
Comment (Gregable) [inline extensions/amp-clamp-text/validator-amp-clamp-text.protoascii]: please replace with amp.dev

File: extensions/amp-clamp-text/0.1/test/validator-amp-clamp-text.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!doctype html>
```
Comment (Gregable) [inline extensions/amp-clamp-text/0.1/test/validator-amp-clamp-text.html]: Please add a copyright statement.

**Comment by sparhami** [comment]
Closing in favor of #22302, addressed above review comments above there.

**Pull request closed without merging**